integration_test/run.sh

```diff
@@ -9,6 +9,7 @@ mkdir -p ${TESTDIR}/1 ${TESTDIR}/2
 killall -9 bitcoind
 
 bitcoind -regtest \
+    -txindex=1 \
```

Comment:
The `txindex` argument was added to the wrong `bitcoind` instance imo. There's another one started a few lines below (line 32, idk why), which is used for testing apparently as the `RPC_URL` env var is set to its port. That's probably why the `0.18.*` tests fail, possibly because `-blockfilterindex=1` is set for all other versions and works for stats too?

Reply:
You are right, the one below has the flag `-server=1` set, while the one above doesn't.
I moved the flag in commit `f7d1f3bce957be592b6f90d1f7667e2a008a1022`.